src/runtime/button-api.js

```diff
@@ -186,32 +199,18 @@ export class ButtonApi {
    * @return {!Element}
    */
   attachSmartButton(deps, button, optionsOrCallback, callback) {
-    const options = /** @type {!../api/subscriptions.SmartButtonOptions} */ (this.getOptions_(
-      optionsOrCallback
-    ));
-    const castedCallback = /** @type {function()} */ (this.getCallback_(
+    const options = this.setupButtonAndGetOptions_(
+      button,
       optionsOrCallback,
       callback
-    ));
-
+    );
     // Add required CSS class, if missing.
     button.classList.add('swg-smart-button');
-    button.addEventListener('click', () =>
-      this.configuredRuntimePromise_.then(configuredRuntime =>
-        configuredRuntime
-          .eventManager()
-          .logSwgEvent(
-            AnalyticsEvent.ACTION_SWG_BUTTON_CLICK,
-            /* isFromUserAction */ true
-          )
-      )
-    );
-
     return new SmartSubscriptionButtonApi(
       deps,
       button,
       options,
-      castedCallback
```

Comment:
Why do we send an empty function here?

Reply:
Because I am manually calling the passed callback here:

  setupButtonAndGetOptions_(button, optionsOrCallback, callbackFun) {
    const options = this.getOptions_(optionsOrCallback);
    const callback = this.getCallback_(optionsOrCallback, callbackFun);
    button.addEventListener('click', event => {
      this.logSwgEvent_(AnalyticsEvent.ACTION_SWG_BUTTON_CLICK, true);
      if (typeof callback === 'function') {
        callback(event);
      }
    });
    return options;
  }

So SmartSubscriptionButtonApi shouldn't have to.  It does make sense to ask this question though: is it possible for someone to click the button in the iframe and not trigger the click event in the code I showed above?